x-pack/plugins/siem/server/lib/timeline/routes/utils/timeline_input.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import {
+  TimelineType,
+  TimelineTypeLiteralWithNull,
+  TimelineSavedObject,
+} from '../../../../../common/types/timeline';
+import { getTimeline, getTemplateTimeline } from './create_timelines';
+import { FrameworkRequest } from '../../../framework';
+
+export class TimelineObject {
+  private id: string | null;
+  private type: TimelineTypeLiteralWithNull;
+  private version: string | number | null;
+  private frameworkRequest: FrameworkRequest;
+
+  public data: TimelineSavedObject | null;
+
+  constructor({
+    id,
+    type = TimelineType.default,
+    version,
+    frameworkRequest,
+  }: {
+    id: string | null;
+    type: TimelineTypeLiteralWithNull;
+    version: string | number | null;
+    frameworkRequest: FrameworkRequest;
+  }) {
+    this.id = id;
+    this.type = type;
+
+    this.version = version;
+    this.frameworkRequest = frameworkRequest;
+    this.data = null;
+  }
+
+  public async getTimeline() {
+    this.data =
+      this.id != null
+        ? this.type === TimelineType.template
+          ? await getTemplateTimeline(this.frameworkRequest, this.id)
+          : await getTimeline(this.frameworkRequest, this.id)
+        : null;
+
+    return this.data;
+  }
+
+  public isExists() {
+    return this.id != null && this.data != null;
+  }
+
+  public isUpdatable() {
+    return this.id != null && this.isExists() && !this.isVersionConflict();
+  }
+
+  public isCreatable() {
+    return !this.isExists();
+  }
+
+  public isUpdatableViaImport() {
+    return this.type === TimelineType.template && this.isUpdatable();
+  }
+
+  public getVersion() {
+    return this.version;
```

Comment:
does it support `template timelines`?

Reply:
yes, timeline_input is a generic input for both timeline and template.
When we init this object for template, we send templateTimelineVersion instead of version
https://github.com/angorayc/kibana/blob/af212242578580adf9e460ec7c4db6dd4d2dd01c/x-pack/plugins/siem/server/lib/timeline/routes/update_timelines_route.ts#L58